### PR TITLE
fix: Unlock PIN after update APN

### PR DESCRIPTION
**Pull request by lwindg**

### CODE REVIEW

File: cellular_utility/management.py [hunk 1]
```diff
@@ -383,6 +386,37 @@ def pdp_context_list(self):
         """Return a list of PDP context."""
         return self._cell_mgmt.pdp_context_list()
 
+    def verify_sim(self):
+        sim_status = self._cell_mgmt.sim_status()
+        _logger.debug("sim_status = " + sim_status.name)
+
+        if sim_status == SimStatus.nosim:
+            self._status = Manager.Status.nosim
+
```
Comment (imZack) [inline cellular_utility/management.py]: return ASAP
```
return sim_status
```

File: cellular_utility/management.py [hunk 2]
```diff
@@ -383,6 +386,37 @@ def pdp_context_list(self):
...
+        elif sim_status == SimStatus.pin:
```
Comment (imZack) [inline cellular_utility/management.py]: `if sim_status == SimStatus.pin:`

File: cellular_utility/management.py [hunk 3]
```diff
@@ -383,6 +386,37 @@ def pdp_context_list(self):
...
+                return sim_status
+
+            # set pin
+            try:
+                self._cell_mgmt.set_pin(self._pin)
+                self._sleep(3, critical_section=True)
+                sim_status = self._cell_mgmt.sim_status()
+                if sim_status == SimStatus.ready:
+                    self._status = Manager.Status.ready
+
```
Comment (imZack) [inline cellular_utility/management.py]: `return sim_status`

File: cellular_utility/management.py [hunk 4]
```diff
@@ -383,6 +386,37 @@ def pdp_context_list(self):
...
+            except CellMgmtError:
+                _logger.warning(format_exc())
+                self._pin = None
+                self._log.log_event_pin_error()
+
+        elif sim_status == SimStatus.ready:
```
Comment (imZack) [inline cellular_utility/management.py]: remove `elif`

Comment (lwindg) [inline cellular_utility/management.py]: remove elif or use if?

Comment (imZack) [inline cellular_utility/management.py]: use `if`

**Pull request merged**